### DISCUSSION TITLE
Update not possible

### DISCUSSION THREAD

**Discussion by PrintModz** [LycheeOrg/Lychee#1954] [Q&A] [answered]
### Detailed description of the problem [REQUIRED]
*A clear and concise description of what the bug is.*

Hello,
I wanted to update the software, but it does not work because of the following error:
****Update not applied: `APP_ENV` in `.env` is `production` and `force_migration_in_production` is set to `0`.****
The hoster wrote that they do not know the error, so I should contact you directly.

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
1. Go to Diagnose
2. Click on Update verfügbar!
3. See error*

**Screenshots**
*If applicable, add screenshots to help explain your problem.*
![2023-07-30 16_26_49-Lychee v4 – Diagnose](https://github.com/LycheeOrg/Lychee/assets/127209437/aa5028cb-4d9a-46d4-9a07-41cdd16965ca)
### Output of the diagnostics  [REQUIRED]
*(Settings => Diagnostics or https://example.com/Diagnostics or `php artisan lychee:diagnostics`)*
Diagnostics
    -----------
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
    Warning: zend.assertions is enabled although Lychee is not in debug mode. Outside of debugging, code generation for assertions is recommended to be disabled for efficiency reasons
    Info: Latest version of PHP is 8.2
    Foreign key: albums.parent_id               → albums.id            : RESTRICT
    Foreign key: albums.id                      → base_albums.id       : RESTRICT
    Foreign key: albums.cover_id                → photos.id            : CASCADE
    Foreign key: base_albums.owner_id           → users.id             : RESTRICT
    Foreign key: photos.album_id                → albums.id            : RESTRICT
    Foreign key: photos.owner_id                → users.id             : RESTRICT
    Foreign key: size_variants.photo_id         → photos.id            : RESTRICT
    Foreign key: sym_links.size_variant_id      → size_variants.id     : RESTRICT
    Foreign key: tag_albums.id                  → base_albums.id       : RESTRICT
    Foreign key: user_base_album.base_album_id  → base_albums.id       : CASCADE
    Foreign key: user_base_album.user_id        → users.id             : CASCADE

### Browser and system
Windows 11

**Comment by ildyria** [comment, marked as answer]
Please go to the _Settings_ page -> at the bottom click on _More_ -> change the value of `force_migration_in_production` to `1` -> go to the bottom and click on _Save_ and reapply your update.

As @d7415  said, this error is very clear.

**Reply by PrintModz** [reply]
INFO Nothing to migrate.

**Comment by PrintModz** [comment]
ahhh Thanks, I didn't know you could change the numbers there. :D

**Comment by PrintModz** [comment]
But now the problem is that it displays the following :  INFO  Nothing to migrate.

**Comment by ildyria** [comment]
You are missing half of the diagnostic. I don't know which version you are on exactly...

**Comment by PrintModz** [comment]
System Information
    --------------
    Lychee Version (release):                4.7.0
    DB Version:                              4.7.0
    
    composer install:                        --no-dev
    APP_ENV:                                 production
    APP_DEBUG:                               false
    
    System:                                  Linux
    PHP Version:                             8.1.16
    PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
    Timezone:                                Europe/Berlin
    Max uploaded file size:                  20M
    Max post size:                           200M
    Max execution time:                      200
    MySQL Version:                           10.5.19-MariaDB-1:10.5.19+maria~ubu2004-log
    
    exec() Available:                        yes
    Imagick Available:                       1
    Imagick Enabled:                         1
    Imagick Version:                         1690
    GD Version:                              2.2.5
    Number of foreign key:                   11 found.

**Comment by ildyria** [comment]
You are using the release version, you need to update the files manually first.

**Comment by PrintModz** [comment]
I have now transferred all the new files to the web space and overwritten them. Now the page is no longer accessible. :D

**Comment by ildyria** [comment]
You should be automatically redirected to a migration page in theory.

**Comment by PrintModz** [comment]
That was exactly the problem, that nothing worked, as you can see above.
